src/app/data_harvester.rs

```diff
@@ -288,17 +293,96 @@ impl DataCollector {
 
         self.update_cpu_usage();
         self.update_memory_usage();
-        self.update_processes();
         self.update_temps();
-        self.update_network_usage();
-        self.update_disks();
-
         #[cfg(feature = "battery")]
         self.update_batteries();
+        #[cfg(feature = "gpu")]
+        self.update_gpus(); // update_gpus before procs for gpu_pids but after temp/batteries/cpu_usage for appending
+        self.update_processes();
+        self.update_network_usage();
+        self.update_disks();
 
         // Update times for future reference.
         self.last_collection_time = self.data.collection_time;
     }
+    #[cfg(feature = "gpu")]
+    #[inline]
+    fn update_gpus(&mut self) {
+        if self.widgets_to_harvest.use_gpu {
+            let use_temp = self.widgets_to_harvest.use_temp;
+            let use_mem = self.widgets_to_harvest.use_mem;
+            let use_proc = self.widgets_to_harvest.use_proc;
+            let use_cpu = self.widgets_to_harvest.use_cpu;
+            let use_battery = self.widgets_to_harvest.use_battery;
+            #[cfg(feature = "nvidia")]
+            if let Some(data) = nvidia::get_nvidia_vecs(
+                &self.temperature_type,
+                &self.filters.temp_filter,
+                use_temp,
+                use_mem,
+                use_proc,
+                use_cpu,
+                use_battery,
```

Comment:
If we do use this big function for it, it might be more convenient to just pass a reference to `self.widgets_to_harvest`.